openthread/Kconfig

```diff
@@ -14,7 +14,7 @@ menu "OpenThread"
 
 choice OPENTHREAD_IMPLEMENTATION
 	prompt "OpenThread origin selection"
-	default OPENTHREAD_SOURCES
+	default OPENTHREAD_LIBRARY
```

Comment:
I believe it will be cleaner to move this selection to `nrf/subsys/net/openthread/Kconfig.defconfig`.

```
choice OPENTHREAD_IMPLEMENTATION
	default OPENTHREAD_LIBRARY
endchoice
```